base/commands/migration/migration_status.go

```diff
@@ -33,10 +33,12 @@ func (s StatusCmd) Exec(ctx context.Context, ec plug.ExecContext) (err error) {
 		return err
 	}
 	defer func() {
-		maybePrintWarnings(ctx, ec, sts.ci, mID.(string))
-		finalizeErr := finalizeMigration(ctx, ec, sts.ci, mID.(string), ec.Props().GetString(flagOutputDir))
-		if err == nil {
-			err = finalizeErr
+		if sts.ci != nil {
+			maybePrintWarnings(ctx, ec, sts.ci, mID.(string))
+			finalizeErr := finalizeMigration(ctx, ec, sts.ci, mID.(string), ec.Props().GetString(flagOutputDir))
```

Comment:
Could you change the `stage.Execute` call above so it uses `""` instead of `any(nil)` as the initial state value?
```
mID, err := stage.Execute(ctx, ec, "", sp)
```
So the state type is `string` and casting it to `string` here is not necessary.